src/autolabel/confidence.py

```diff
@@ -162,7 +169,7 @@ def logprob_average_per_key(
 
         # Find the locations of each key in the logprobs as indices
         # into the logprobs list
-        locations = []
```

Comment:
comment on why we need this?

Reply:
This is just implementation choice - allows to avoid setting `logprob_per_key[locations[-1][2]]` at the end explicitly.